### PR TITLE
Implement an "Ethereum Address" react component

### LINKED ISSUE #315: Implement an "Ethereum Address" react component
This component should accept both addresses and ENS names, and trigger events when modified and when a valid value is entered. It should have the following features:

 - Accepts a web3 `provider` object with which to do ENS lookups.
 - When an address that has a reverse record is entered, show the name alongside the address. 
 - When an ENS name is entered, resolve it and show the address alongside the name.
 - Visually indicate to users whether the current input is valid or not.
 - Support styling via `className` prop.
 - Support an optional 'visual checksum' such as identicons or blockies.
 - Include documentation specifying how to use it and how to style it.

### PR DESCRIPTION
**Pull request by industral**
fixed #315

### CODE REVIEW

**Comment by industral** [comment]
Screen Shot 2019-05-24 at 7 15 30 PM

**Comment by industral** [comment]
[E2E tests Video](https://www.dropbox.com/s/q21hii38fq9x80e/eth-address-e2e.mov?dl=0)
[Demo Video](https://www.dropbox.com/s/1hi3kjt4kwudtmy/eth-address-demo.mov?dl=0)

**Comment by Arachnid** [comment]
This looks pretty good to me.

At the moment, the blockie returned depends on the value the user entered into the field. It needs to depend only on the address - so if the user enters an address, use that; if they enter a name, resolve it and use the address. In either case, it needs to be lower-cased.

It'd also be useful if the event handler gave the containing component access to both the name and the address, where they're known.

**Comment by jefflau** [comment]
Hi @industral,

Can you refactor from the class components to hooks as you have used for the cypress Test route?

**Comment by industral** [comment]
@Arachnid You right. Fixed it and add test as well.
@jefflau done.

File: src/components/Address/Address.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+const ENS_NOT_FOUND = 'ENS name not found'
+
+function Address(props) {
+  const [resolvedAddress, setResolvedAddress] = useState(null)
+  const [inputValue, setInputValue] = useState('')
+  const [isResolvingInProgress, setIsResolvingInProgress] = useState(false)
+  const [error, setError] = useState(null)
+
+  const ens = new ENS(props.provider)
```
Comment (jefflau) [inline src/components/Address/Address.js]: It's probably not a good idea to reinstantiate this every time we render. We should probably only do this once

Comment (industral) [reply src/components/Address/Address.js]: You're right. Fixed.

**Comment by jefflau** [comment]
Hi @industral,

Sorry for the late reply. I had our designer whip up some designs for how this should actually look in practice. The design you used was for our search bar, but we're more likely using this for our inputs in when editing the resolver/owner etc...

Screen Shot 2019-06-05 at 8 24 16 PM
Screen Shot 2019-06-05 at 8 20 28 PM

**Comment by industral** [comment]
Hi @jefflau!  Let me clarify. So basically I need to change only height of component, to reflect 2 states: when empty and when filled. And add green checkmark as well?

**Comment by jefflau** [comment]
@industral In addition to that, to match the styles. The designer's comments about 42 height empty vs filled is regarding our current `Input` component. Your component would essentially replace this for addresses.

You can check the `Input` component, which has all the styles. If the input is an address, the check mark is to show it is a valid address, not that it found a reverse record. If the input is a name, the check mark will be to show it this name resolves to an address. Essentially the check mark is for validation. The blockies will show up automatically when a correct address is show.

**Comment by industral** [comment]
Hey guys! Sorry for late response. I'm now on full-time job, not sure I'll have time at near future to look into that issue. I'm drop off from bounty, feel free to use that PR.

**Comment by jefflau** [comment]
Merged this into a separate branch to do additional work. Closing this PR

**Pull request closed without merging**